wcfsetup/test.php

```diff
@@ -347,18 +347,16 @@ function checkResult() {
 	foreach ($requiredExtensions as $extension) {
 		if (!extension_loaded($extension)) return false;
 	}
-	
-	if (!extension_loaded('gd') && !extension_loaded('imagick')) return false;
 
-	if (extension_loaded('imagick') && !\in_array('WEBP', \Imagick::queryFormats())) {
-		return false;
+	if (extension_loaded('imagick') && \in_array('WEBP', \Imagick::queryFormats())) {
+		return true;
 	}
 
-	if (extension_loaded('gd') && empty(\gd_info()['WebP Support'])) {
-		return false;
+	if (extension_loaded('gd') && !empty(\gd_info()['WebP Support'])) {
+		return true;
 	}
 	
-	return true;
+	return false;
```

Comment:
I don't like the logic inversion here, because one can't easily add additional conditions at the bottom, as a successful check for the image processing library will return true.

I guess adding a `$imageProcessingWorks = true;` in place of the `return true` does the job most cleanly.